valkyrie/lib/valkyrie/persistence/fedora/query_service.rb

```diff
@@ -31,10 +31,12 @@ def include_uris
       ]
     end
 
-    def find_members(resource:)
-      Array(resource.member_ids).lazy.map do |id|
+    def find_members(resource:, model: nil)
+      result = Array(resource.member_ids).lazy.map do |id|
         find_by(id: id)
       end.select(&:present?)
+      return result unless model
```

Comment:
I'm seeing a likely notable Ruby in-memory performance issue here.

If model is given, you'll be iterating at least three times through results:
* map (and find_by)
* select (on present?)
* select (on is_a?(model))

If the model is not given, you'll be iterating at least  two times through results:
* map (and find_by)
* select (on present?)

Reply:
I agree with your assessment but I'm not concerned about optimizing right now. I think that can be done at a later point if this is indeed a hotspot.

Reply:
Really? I thought the lazy iteration would take care of that.